packages/mui-material/src/FormHelperText/FormHelperText.js

```diff
@@ -110,15 +110,17 @@ const FormHelperText = React.forwardRef(function FormHelperText(inProps, ref) {
     required: fcs.required,
   };
 
+  delete ownerState.ownerState;
+
   const classes = useUtilityClasses(ownerState);
 
   return (
     <FormHelperTextRoot
       as={component}
-      ownerState={ownerState}
       className={clsx(classes.root, className)}
       ref={ref}
       {...other}
+      ownerState={ownerState}
```

Comment:
@mnajdova @siriwatknp I want to bring your attention to this change. There are similar ones in this PR, as well as similar PRs, for example, [this one](https://github.com/mui/material-ui/pull/41903).

With the standardization of the slots pattern, we've run into this multiple times:

- Parent component has `ownerState`, let's call it `ownerStateParent`
- Child slot component has it's own `ownerState`, let's call it `ownerStateChild`, that it needs to provide to its own children

The `ownerStateParent` is [provided to the child slot component](https://github.com/mui/material-ui/blob/fd764b5578ef6688f3c9c1a12d986de9d938de8d/packages/mui-material/src/utils/useSlot.ts#L142), as it's useful when overriding.

But the child needs to take care of two things:

- That the `ownerStateParent`, coming through props, doesn't override its own `ownerStateChild` when passing it to its own children. This is the reason for the change on [lines 118-123](https://github.com/mui/material-ui/pull/42062/files#diff-eb5bb0f7d6fd1be741c79c303d033419ee5eefea4124632e06c5b0f4bcf5c0d4R119-R124): `ownerState={ownerState}` must come after `{...other}`.
- That the `ownerStateParent` is not nested inside its own `ownerStateChild`, otherwise, it would override `ownerStateChild` because of how `createStyled` merges props ([reference](https://github.com/mui/material-ui/blob/fd764b5578ef6688f3c9c1a12d986de9d938de8d/packages/mui-system/src/createStyled/createStyled.js#L69)). This could cause bugs when accessing the `ownerState` in style variants props matching. Like it's done in this component ([line 68](https://github.com/mui/material-ui/pull/42062/files#diff-eb5bb0f7d6fd1be741c79c303d033419ee5eefea4124632e06c5b0f4bcf5c0d4R68)). This is the reason for `delete ownerState.ownerState` on this file.

The pattern of spreading `other` into the root component, as well as spreading `props` into `ownerState`, is spread throughout the codebase, and that's why we've been bumping into this repeatedly. This behavior is error-prone, as it's not evident unless you're familiar with the codebase.

I think it might be worth it to invest in a shared mechanism that all components can rely on to safely spread `other` and/or `props` without propagating unwanted `ownerState`s.

What do you think?

Reply:
Yes, we should have a single place that takes care of that logic. I spent more than an hour investigating `ownerState` issues every time I came across it.

To be honest, this is one thing we should clean up in v7. I already took an approach for Joy UI to never inherit other component which turns out to be easier to maintain. The `ownerState` pass-through is predictable and reliable.

Always inherit styled-component (to get styles) from another component instead of wrapping the whole component. Duplicated logic might happen but it's fine compare to the time we have to fix bugs.

```js
// This is what I prefer
// Paper.js
export const StyledPaperRoot = styled('div')( … );

const Paper = (...) => (
  return <StyledPaperRoot ...> 
)

// AppBar.js
import { StyledPaperRoot } from '../Paper/Paper'; // The styled-component is used, not the `Paper`

const AppBarRoot = styled(StyledPaperRoot)(…)

const AppBar = () => {
  return <AppBarRoot ...>
}
```

Reply:
Don't we have a `useSlotProps` that can abstract this in one place? I know we had it in both Base UI and Joy UI.

On the comment of Jun, I always had a mixed feelings honestly. The styles often depend on the props of the component, expecting the Paper has some props that exist in the `ownerState`, so I would much rather use the actual component than the styled root. If we have only common styles shared, we can share the styles only, not necessarily the component.

Reply:
Thank you for your replies! I wondered if I had a misunderstanding, but I'm glad we're on the same page about this being a problem 🙌🏼 

Regarding solutions, after the slot pattern is standardized throughout Material UI, we should invest time in improving its infrastructure (e.g., avoiding the issue) and improving the documentation around it.

> To be honest, this is one thing we should clean up in v7.

Yes, after v6 is stable we can discuss if a component restructure would make sense 🙌🏼 

> Don't we have a `useSlotProps` that can abstract this in one place?

Yes, we can abstract it there. Although for that to work, all child components would have to use the hook, not all do so now. It could make sense for all components to have at least the `root` slot.

I will create a couple of issues in which we can start collecting this work.

Reply:
Created issues:

- https://github.com/mui/material-ui/issues/42183
- https://github.com/mui/material-ui/issues/42184